src/loadgenerator/loadgenerator-entrypoint.sh

```diff
@@ -24,4 +24,13 @@ if [[ -z "${FRONTEND_ADDR}" ]]; then
 fi
 
 set -x
-locust --host="http://${FRONTEND_ADDR}" "$@"
+
+# if one request to the frontend fails, then exit
+STATUSCODE=$(curl --silent --output /dev/stderr --write-out "%{http_code}" http://${FRONTEND_ADDR})
+if test $STATUSCODE -ne 200; then
+    echo "Error: Could not reach frontend - Status code: ${STATUSCODE}"
+    exit 1
+fi
+
+# run locust
+locust --host="http://${FRONTEND_ADDR}" --web-port=$PORT
```

Comment:
I don't see a default port being set anywhere. It will be auto-injected by Cloud Run, but it's a good idea to have a default value for re-usability

I believe you can set a PORT value in the Dockerfile that will be overwritten at runtime. Or you can do a null check in this script